### PR TITLE
Update README and CHANGELOG for Event Hubs

### PR DESCRIPTION
**Pull request by srnagar**
In preparation for Preview 4 release, updating README and CHANGELOG for Event Hubs.

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Release History
 
+## 5.0.0-preview.4 (2019-10-08)
+- Proxy support for Event Hub Clients.
+- EventHubConsumer now provides last enqueued event information.
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md]: the async client also provides this.

File: sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
+- Refactored `azure-messaging-eventhubs` to extract AMQP implementation details to `azure-core-amqp` module.
+- Added modules support for JDK 9+.
+- Renamed model classes to support Java bean naming convention.
+- EventHubClient now provides method to get the name of the Event Hub associated with the client.
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md]: The async client supports this too.

File: sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md (hunk 1 above)
Comment (srnagar) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md]: Updated.

File: sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md (hunk 2 above)
Comment (srnagar) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md]: Updated

**Pull request merged**